docs/manual/install.qmd

```diff
@@ -0,0 +1,89 @@
+# Installation
+
+The commands below can be entered using any terminal application. Search for
+"terminal" or "command line" in your OS application menu.
+
+**Attention**: the python and pip executables may have different names
+depending on the distribution. `python` and `pip` is used below, but they may
+also be called `python3` and `pip3`.
+
+## System dependencies
+
+- Python >= 3.8 (see @sec-python)
+- SQLite 3 (see @sec-sqlite)
+
+## Package
+
+Install the package:
+
+`pip install git+https://github.com/Big-Life-Lab/PHES-ODM-sharing.git@impl`
+
+(The `impl` branch must be used for now, until it's merged into the main
+branch.)
+
+You're done! You can now explore how to get started in @sec-getting-started.
+
+## Development
+
+If you want to setup a development environment, then run the commands below as
+you see fit. If you just want to use the library, then you can ignore this
+section.
+
+```bash
+# clone the repo
+git clone https://github.com/Big-Life-Lab/PHES-ODM-sharing odm-sharing
+cd odm-sharing
+
+# if python != 3.8, then install pyenv and setup repo to run python 3.8
+sudo apt-get install pyenv
+pyenv install 3.8
+pyenv local 3.8
+
+# create a virtual env for the repo, using the selected python version
+pyenv exec python -m venv .env # with pyenv
+#python -m venv .env # without pyenv
+
+# activate the virtual env (must be done every time a new terminal is opened,
+# to set the correct python-binary path/env)
+source .env/bin/activate
+
+# verify that python now has the correct version (3.8)
+python --version
+
+# make sure pip is up to date, and install dependencies
+pip install --upgrade pip
+pip install -r requirements.txt
+pip install -r requirements-dev.txt
+
+# if you previously installed the package, then uninstall it again
+pip uninstall odm_sharing
+
+# build and install the library in development mode
+pip install -e .
+```
+
+You can now edit the source files, and directly see the changes when importing
+the package.
+
+## Documentation
+
+To generate the documentation, you'll first have to:
+
+- Install [Quarto](https://quarto.org/docs/get-started/)
+- Install Python dependencies:
+
+    ```bash
+    pip install -r requirements-doc.txt
+    ```
+
+To preview the manual in a web browser while editing:
+
+```bash
+quarto preview docs/manual
+```
+
+To render the manual to a PDF:
+
+```bash
+quarto render docs/manual --to pdf
```

Comment:
done. It's now rendered to `docs/manual` (since that's what using `.` in `_quarto.yml` does, and I didn't want to use `../../`).

Reply:
Any reason for not putting it in the root (i.e not wanting to use ../../)?

Reply:
yeah, one objective reason would be that if you're already in `docs/manual` and run `quarto render` (which renders the project in the current directory by default), then the output wouldn't end up where you expect (the current project dir).

Reply:
I think that most people would be running the command from the root of the project and not cd'ing into the docs directory. Also, when building the docs version of the website a lot of files are added which would be annoying to ignore if they're not in a directory.

Reply:
I see it like this: the manual is contained within the `docs/manual` directory, and so I don't think it makes sense to put the output outside of that dir. I also think it's intuitive enough that the output ends up in `docs/manual` when the user runs `quarto render docs/manual`, don't you think so too? I'm not sure what you mean regarding the ignored files.

Reply:
1. I guess it wasn't intuitive to me. I guess its fine there but we should update the documentation saying where to find it after running the command.
2. When running the command to build the website with `quarto render docs/manual --to html` it adds the built HTML files to the docs/manual which should be gitignored.

Reply:
1. I agree with you that when running a command in a certain directory you'd expect the output there as well, but I don't think that's possible here (since we have to specify an absolute path, like the repo root), and in this case we're explicitly giving the path of the quarto project when running the render command, also, when working with a project it's normal that the files are local to the project. I've already mentioned the output dir in the docs, right under the command itself, so it should be good.
2. Good catch, thank you. Fixed now. I think this is another good reason to keep the output in the project dir, so that it doesn't pollute the repo root.

Reply:
done, everything is output to the build dir